upstream: yes

From 177fda86451cdeaaea8ed409e6d711b670699a97 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ga=C3=ABl=20Bonithon?= <gael@xfce.org>
Date: Tue, 6 Feb 2024 18:14:04 +0100
Subject: [PATCH 1/2] screen: Fix wrong assert

It's always been wrong (or has been for a long time) but de3e7aac
revealed it, because now it's no longer disabled by building with
--disable-debug.

Fixes: de3e7aac72fdcd3e62d69f37ec2570e5d668950a
Closes: #299
---
 terminal/terminal-screen.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/terminal/terminal-screen.c b/terminal/terminal-screen.c
index 6e48b522..dc931ec7 100644
--- a/terminal/terminal-screen.c
+++ b/terminal/terminal-screen.c
@@ -1892,7 +1892,7 @@ terminal_screen_paste_unsafe_text (TerminalScreen *screen,
 {
   GtkWidget *dialog;
 
-  g_return_if_fail (original_clipboard != GDK_SELECTION_CLIPBOARD && original_clipboard != GDK_SELECTION_PRIMARY);
+  g_return_if_fail (original_clipboard == GDK_SELECTION_CLIPBOARD || original_clipboard == GDK_SELECTION_PRIMARY);
 
   dialog = terminal_screen_unsafe_paste_dialog_new (screen, text);
   gtk_widget_show_all (dialog);
-- 
2.43.2


From 71ecd2a721b059394edcfb316a20ac9d46a7b2cd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ga=C3=ABl=20Bonithon?= <gael@xfce.org>
Date: Wed, 7 Feb 2024 16:10:05 +0100
Subject: [PATCH 2/2] prefs-dialog: Fix wrong assert

Related: #299
Fixes: de3e7aac72fdcd3e62d69f37ec2570e5d668950a
Closes: #300
---
 terminal/terminal-preferences-dialog.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/terminal/terminal-preferences-dialog.c b/terminal/terminal-preferences-dialog.c
index 86950b45..a1671400 100644
--- a/terminal/terminal-preferences-dialog.c
+++ b/terminal/terminal-preferences-dialog.c
@@ -1893,7 +1893,7 @@ terminal_preferences_dialog_presets_changed (TerminalPreferencesDialog *dialog,
   GValue        src = { 0, };
   GValue        dst = { 0, };
 
-  g_return_if_fail (TERMINAL_IS_PREFERENCES_DIALOG (widget));
+  g_return_if_fail (TERMINAL_IS_PREFERENCES_DIALOG (dialog));
   g_return_if_fail (GTK_IS_COMBO_BOX (widget));
 
   combobox = GTK_COMBO_BOX (widget);
-- 
2.43.2

